### ISSUE TITLE
Shadowsocks-libev 3.0.8 cannot start or work

### ISSUE THREAD

**Issue by jkr978** [shadowsocks/shadowsocks-libev#1606]
(Please mention that if the issue you filed is solved, you may wish to close it by yourself. Thanks again.)

### What version of shadowsocks-libev are you using?
3.0.8

### What operating system are you using?
OS X 10.10.5

### What did you do?
brew upgrade shadowsocks-libev
brew services start shadowsocks-libev
vi ~/.shadowsocks.json
ss-local -c ~/.shadowsocks.json

### What did you see instead?
 2017-07-31 21:30:06 INFO: initializing ciphers... aes-256-cfb
 2017-07-31 21:30:06 ERROR: bind: Address already in use
 2017-07-31 21:30:06 ERROR: bind() error

### What is your config in detail (with all sensitive info masked)?
{
   "server":"xx.xxx.xxx.xxx",
   "server_port":xxxx,
   "local_address": "127.0.0.1",
   "local_port":1080,
   "password":"xxxxxx",
   "timeout":300,
   "method":"aes-256-cfb"
}

The process listening to port 1080 is "ss-local". I tried to kill it. But it keeps restarting with a new PID number and cannot be killed. 
The parent process of "ss-local" is "launchd(1)" under "kernel-task(0)". If I quit "kernel-task" my computer will be rebooted, but the error is still here.

I apologize if this is basic, but I know nothing about computer. Google doesn't help this time.
Any discussion appreciated.

**Comment by realmirai** [comment]
Though I've never used a Mac, I think it is what called a 'launchctl' that has been starting the service.
You can try `brew services stop shadowsocks-libev` and the process may quit. Then you are able to start ss-local manually.
You can also configure the service so the system will load ss-local every time you log in.
It's said the service's default config file is at `/usr/local/etc/shadowsocks-libev.json`

**Comment by jkr978** [comment]
@realmirai 
Thanks. `brew services stop shadowsocks-libev` solved the problem.

However, shadowsocks isn't working. Every time I try to connect Google it goes like:
2017-08-01 20:53:31 ERROR: getpeername: Invalid argument

I checked the server. Netstat shows two sshd is listening.
Active Internet connections (only servers)
Proto Recv-Q Send-Q Local Address               Foreign Address             State       PID/Program name   
tcp        0      0 0.0.0.0:abcde               0.0.0.0:*                   LISTEN      957/sshd            
tcp        0      0 :::abcde                    :::*                        LISTEN      957/sshd            

Is there a problem with the server? 
Thanks again.

**Comment by jkr978** [comment]
However, shadowsocks isn't working. Every time I try to connect Google it goes like:
2017-08-01 20:53:31 ERROR: getpeername: Invalid argument

I checked the server. Netstat shows two sshd is listening.
Active Internet connections (only servers)
Proto Recv-Q Send-Q Local Address               Foreign Address             State       PID/Program name   
tcp        0      0              0.0.0.0:abcde               0.0.0.0:*                   LISTEN      957/sshd            
tcp        0      0                       :::abcde                    :::*                        LISTEN      957/sshd           
 
I also installed shadowsocks-libev in my server, just the day before its latest update. 
According to https://github.com/shadowsocks/shadowsocks-libev/issues/804 I add `"nameserver": "8.8.8.8",` in my server config.

And then it works well.
